isisd/isis_te.c

```diff
@@ -111,9 +113,13 @@ void isis_mpls_te_create(struct isis_area *area)
 	if (area->mta->ted)
 		isis_te_init_ted(area);
 
-	/* Update Extended TLVs according to Interface link parameters */
-	for (ALL_LIST_ELEMENTS_RO(area->circuit_list, node, circuit))
+	/* Update Extended TLVs according to Interface link parameters
+	 * and neighbor IP addresses
+	 */
+	for (ALL_LIST_ELEMENTS_RO(area->circuit_list, node, circuit)) {
 		isis_link_params_update(circuit, circuit->interface);
+		isis_mpls_te_circuit_ip_update(circuit);
+	}
```

Comment:
You should not proceed like that. The call to `isis_mpls_te_circuit_ip_update()` will trigger an `lsp_regenerate_schedule()`. But, `isis_mpls_te_create()`  is called by `isis_instance_mpls_te_create()` which at the end call also an  `lsp_regenerate_schedule()`. In addition, if an interface has both an IPv4 and IPv6 addresses, you will call at the end 3x the LSP regeneration.

I think it will safer to copy/paste back in `isis_link_params_update()` part of the code from `isis_mpls_te_adj_ip_enabled()' but, we'll loose the global information about IPv6 address.

Reply:
I have found another working method to avoid multiple LSP regeneration.

Let me know if you are happy with it